src/include/migraphx/op/allocate.hpp

```diff
@@ -36,20 +36,52 @@ namespace op {
 struct allocate
 {
     shape s{};
+    // for dynamic allocate to set the buffer type
+    shape::type_t buf_type = shape::half_type;
+    // flag for dynamic allocate to use shape attr rather than [0, max] ranges in compute_shape()
+    // useful for optimizations later on
+    bool use_shape_attr = false;
```

Comment:
This should store the dynamic dymension instead of using a boolean. It should be `std::vector<dynamic_dimensions>` and then if its not empy then use those dynamic dimensions for the shape.

Reply:
What difference does that make versus using the shape attribute?